tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -9060,7 +9060,7 @@ public function dataGeneralizeScope(): array
 	{
 		return [
 			[
-				'array<non-empty-array<int|string, array{saveCount: int<0, max>, removeCount: int<0, max>, loadCount: int<0, max>, hitCount: int<0, max>}>>',
+				'array<string, non-empty-array<int|string, array{saveCount: int<0, max>, removeCount: int<0, max>, loadCount: int<0, max>, hitCount: int<0, max>}>>',
```

Comment:
tbh, I don't know why this test got more precise with this change :)

Reply:
Because preg_replace is impacted by your changes. By using selectFromArgs, conditional return type is resolved.